ledger/roundlru.go

```diff
@@ -121,7 +121,7 @@ func (hlc *heapLRUCache) inc() {
 	}
 }
 func (hlc *heapLRUCache) reIndex() {
-	if hlc.entries.heap != nil || len(hlc.entries.heap) == 0 {
+	if hlc.entries.heap == nil || len(hlc.entries.heap) == 0 {
```

Comment:
that's a good find. However, I think that you can simplify it to 
```golang
if len(hlc.entries.heap) == 0 {
```
since `len(nil)` is zero.